Cygwin/X: Allow pointer warping to work in rootless modes

Mouse pointer warping in multiwindow/rootless mode was never implemented,
due to concerns that moving the mouse pointer without asking might be rude

This patch allows X applications to move the mouse pointer in rootless modes,
Let's hope they don't abuse this privilege ;-)

---
 xserver/hw/xwin/wincursor.c |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

Index: xorg-server-1.5.3/xserver/hw/xwin/wincursor.c
===================================================================
--- xorg-server-1.5.3.orig/xserver/hw/xwin/wincursor.c	2008-11-05 16:52:16.000000000 +0000
+++ xorg-server-1.5.3/xserver/hw/xwin/wincursor.c	2009-01-12 19:42:54.000000000 +0000
@@ -99,8 +99,16 @@
       return;
     }
 
-  /* Only update the Windows cursor position if we are active */
-  if (pScreenPriv->hwndScreen == GetForegroundWindow ())
+  /*
+     Only update the Windows cursor position if root window is active,
+     or we are in a rootless mode
+  */
+  if ((pScreenPriv->hwndScreen == GetForegroundWindow ())
+      || pScreenPriv->pScreenInfo->fRootless
+#ifdef XWIN_MULTIWINDOW
+      || pScreenPriv->pScreenInfo->fMultiWindow
+#endif
+      )
     {
       /* Get the client area coordinates */
       GetClientRect (pScreenPriv->hwndScreen, &rcClient);
